students/jonathandvu/lesson04/mailroom2.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env python
+# PY210 Lesson 4 Mailroom Assignment
+# Jonathan Vu
+#
+# Properties
+import os
+from datetime import date
+
+donor_donations = {"Bill Gates": [85, 100],
+                   "Jeff Bezos": [65, 25, 55],
+                   "Elon Musk": [15, 25],
+                   "The Rock": [250, 125, 55],
+                   "Kevin Hart": [30]}
+
+
+# Support Functions
+def get_donor_names():
+    for x in donor_donations:
+        print('{}'.format(x))
+
+def send_a_thank_you():
+    print('\n----- Send a Thank You -----')
+    prompt = "\n".join(('Please select from the options:',
+                        '1: Type in the full name of a donor.',
+                        '2: Type in ''list'' to see donors.',
+                        '3: Type in a new donor name',
+                        '>> '))
+    user_action_TY = input(prompt)
+    donor_name_list = list(donor_donations.keys())
+
+    if user_action_TY == 'list':
+        get_donor_names()
+        send_a_thank_you()
+    else:
+        donation = input('Donation Amount ($): ')
+        try:
+            donation = int(donation)
+        except ValueError:
+            print('Enter a numeric value!')
+            donation = int(donation)
+
+        if user_action_TY in donor_name_list:
+            for donors in donor_name_list:
+                if donors == user_action_TY:
+                    donor_donations[user_action_TY].append(donation)
+        elif user_action_TY not in donor_donations.keys():
+            donor_donations[user_action_TY] = [donation]
```

Comment:
don't loop over donor - utilize dictionary to look up your donor -> `donor_donations.get(user_action_ty)`